core/slim_sim.cpp

```diff
@@ -8345,12 +8347,6 @@ void SLiMSim::__ConfigureSubpopulationsFromTables(EidosInterpreter *p_interprete
 		// validate and parse metadata; get metadata values or fall back to default values
 		size_t metadata_length = pop_table.metadata_offset[pop_index + 1] - pop_table.metadata_offset[pop_index];
 		
-		if (metadata_length == 0)
```

Comment:
This interacts with the struct-to-JSON translation code in _InstantiateSLiMObjectsFromTables() at slim_sim.cpp:9025; if it sees a zero-length struct metadata chunk, it leaves it zero-length, which then hits this case here and gets treated as empty.  So if you remove this case here (which seems like the right change), then in the struct-to-JSON translation code you will need to write out the 4-byte "null" rather than just leaving it empty.  I will fix that with my own commit after accepting this PR, to keep the ball rolling forward; just wanted you to be aware of the existence of that new translation code.